src/WebJobs.Extensions/Extensions/Timers/Listener/TimerListener.cs

```diff
@@ -209,14 +215,25 @@ private void OnTimer(object sender, ElapsedEventArgs e)
                 // schedule. Errors will be logged to Dashboard already.
             }
 
+            // If the trigger fired before it was officially scheduled (likely under 1 second due to clock skew),
+            // adjust the invocation time forward for the purposes of calculating the next occurrence.
+            // Without this, it's possible to set the 'Next' value to the same time twice in a row, 
+            // which results in duplicate triggers if the site restarts.
+            DateTime adjustedInvocationTime = invocationTime;
+            if (!isPastDue && !runOnStartup && ScheduleStatus?.Next > invocationTime)
```

Comment:
I was planning on having this be the first package released from the AppVeyor build, so it should be okay.

---
In reply to: [73402013](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/133#discussion_r73402013) [](ancestors = 73402013)